shell/app/modules/application/common/components/health-check-field.tsx

```diff
@@ -11,188 +11,176 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program. If not, see <http://www.gnu.org/licenses/>.
 
-import React, { PureComponent } from 'react';
+import React from 'react';
 import { Input, InputNumber, Select } from 'app/nusi';
+import { useUpdate } from 'common';
 import i18n from 'i18n';
 import './variable-input-group.scss';
 
 const { Option } = Select;
 
 interface IVariableInputGroupProps {
-  value: any;
-  placeholder?: string;
+  value: Obj;
   disabled?: boolean;
-  onChange: (options: any[]) => void;
+  onChange: (val: Obj) => void;
 }
 
 const defaultValue = {
   healthCheckKey: null,
 };
 
-export default class extends PureComponent<IVariableInputGroupProps, any> {
-  constructor(props: IVariableInputGroupProps) {
-    super(props);
-    const keys = Object.keys(props.value);
-    let healthCheckKey;
-    if (keys.length) {
-      healthCheckKey = keys[0];
-    }
-    this.state = {
-      healthCheckKey,
-      value: props.value || defaultValue,
-    };
-  }
+const HealthCheckField = (props: IVariableInputGroupProps) => {
+  const { value: propsValue, disabled, onChange } = props;
+  const [{ healthCheckKey, value }, updater, update] = useUpdate({
```

Comment:
seems updater is not using, u can write as `const [{ healthCheckKey, value }, , update]` to avoid warning